### PR TITLE
Added new dependency to CGO usb bindings

### PR DESCRIPTION
**Pull request by likewhoa**
This is in relation to bug 547204 at b.g.o which bug 523990 needs in order to properly compile. I will send another PR and reference it which will contain the ebuild for dev-libs/go-usb

### CODE REVIEW

**Comment by likewhoa** [comment]
Commit for dev-libs/go-usb package. Please review.

**Comment by mgorny** [comment]
Please leave comment on the bug when you submit a PR, so the assignee knows about it :P.

File: dev-libs/go-usb/go-usb-9999.ebuild [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+	for i in LICENSE *.go; do
+		cp "${S}/${i}" "${D}/${GO_PN}" || die "Install failed"
+	done
+}
+
+src_test() {
+	go test -ldflags '-extldflags=-fno-PIC' ${GO_PN} || die
+}
+
```
Comment (mgorny) [inline dev-libs/go-usb/go-usb-9999.ebuild]: Stray empty line.

File: dev-libs/go-usb/go-usb-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+# Tests require a connected mtp device
+RESTRICT="test"
+
+GO_PN="/usr/$(get_libdir)/go/src/github.com/hanwen/usb"
+
+src_install() {
+	dodir "${GO_PN}"
+
+	rm -r "${S}/.git" || die "Failed to remove .git directory"
```
Comment (mgorny) [inline dev-libs/go-usb/go-usb-9999.ebuild]: Why is this necessary?

File: dev-libs/go-usb/go-usb-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2014 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Header: $
+
+EAPI=5
+
+inherit git-r3 flag-o-matic multilib toolchain-funcs
```
Comment (mgorny) [inline dev-libs/go-usb/go-usb-9999.ebuild]: Most of them unused :P.

File: dev-libs/go-usb/go-usb-9999.ebuild (hunk 1 above)
Comment (likewhoa) [inline dev-libs/go-usb/go-usb-9999.ebuild]: I'll fix this and those below during the week. Sorry for late reply, it seems i wasn't getting these email notifications.

File: dev-libs/go-usb/go-usb-9999.ebuild (hunk 2 above)
Comment (likewhoa) [inline dev-libs/go-usb/go-usb-9999.ebuild]: I removed most of that now, do review.

File: dev-libs/go-usb/go-usb-9999.ebuild (hunk 3 above)
Comment (likewhoa) [inline dev-libs/go-usb/go-usb-9999.ebuild]: Yea, this was a copy from go-mtpfs but i cleaned it up now.

File: dev-libs/go-fuse/go-fuse-9999.ebuild [hunk 4]
```diff
@@ -46,7 +46,7 @@ done
 
 insinto /usr/lib/go/
 doins -r "${S}/pkg"
-insinto "/usr/lib/go/src/pkg/${GO_PN}/"
+insinto "/usr/lib/go/src/${GO_PN}/"
```
Comment (mgorny) [inline dev-libs/go-fuse/go-fuse-9999.ebuild:49]: I've pinged go-fuse maintainer about this.

File: dev-libs/go-usb/go-usb-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+KEYWORDS=""
+IUSE=""
+
+DEPEND=">=dev-lang/go-1.4"
+RDEPEND=""
+
+# Tests require a connected mtp device
+RESTRICT="test"
+
+GO_PN="/usr/$(get_libdir)/go/src/github.com/hanwen/usb"
```
Comment (mgorny) [inline dev-libs/go-usb/go-usb-9999.ebuild]: Are you sure it's supposed to be libdir and not plain `lib`? go-fuse above uses `lib`. This is important for systems where lib!=lib64.

File: dev-libs/go-usb/go-usb-9999.ebuild [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+src_install() {
+	insinto "${GO_PN}"
+
+	for i in LICENSE *.go; do
+	  doins "${i}" || die "Install failed"
```
Comment (mgorny) [inline dev-libs/go-usb/go-usb-9999.ebuild]: `doins` can take multiple files. No need to loop.

File: dev-libs/go-usb/go-usb-9999.ebuild (hunk 5 above)
Comment (likewhoa) [inline dev-libs/go-usb/go-usb-9999.ebuild]: I am not sure actually so perhaps it should be removed.

File: dev-libs/go-usb/go-usb-9999.ebuild (hunk 6 above)
Comment (likewhoa) [inline dev-libs/go-usb/go-usb-9999.ebuild]: The problem was that doing 'doins' by itself was putting it into ${D}/${PF} for some reason. I will test again to confirm.

**Comment by likewhoa** [comment]
@ZeroChaos- see my last commit, i removed the added get_libdir and doins loop since it's not even necessary.

**Comment by ZeroChaos-** [comment]
still approved

**Comment by mgorny** [comment]
Ok, checked this and go itself uses /usr/lib.

File: dev-libs/go-usb/go-usb-9999.ebuild [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+RDEPEND=""
+
+# Tests require a connected mtp device
+RESTRICT="test"
+
+GO_PN="/usr/lib/go/src/github.com/hanwen/usb"
+
+src_install() {
+	insinto "${GO_PN}"
+	doins *.go LICENSE || die "Install failed"
```
Comment (mgorny) [inline dev-libs/go-usb/go-usb-9999.ebuild:28]: die unnecessary in EAPI 5 (with doins).

**Comment by mgorny** [comment]
Ok, I'm going to fix that tiny bug and merge it.

**Comment by mgorny** [comment]
All merged now. Closing.
